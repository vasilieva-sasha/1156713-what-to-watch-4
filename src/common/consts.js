export const TIMEOUT = 1000;

export const SIMILAR_FILMS_AMOUNT_SHOW = 4;

export const ALL_GENRES = `All genres`;

export const MAX_GENRES = 9;

export const FILMS_SHOW_AMOUNT = 8;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const CurrentPage = {
  MAIN: `MAIN`,
  INFO: `INFO`,
  LOGIN: `LOGIN`,
  REVIEW: `REVIEW`,
  MYLIST: `MYLIST`
};

export const AppRoute = {
  MAIN: `/`,
  SIGN_IN: `/login`,
  MYLIST: `/mylist`,
  FILM: `/films`,
  ADD_REVIEW: `/review`,
  PLAYER: `/player/`
};

export const RATING = 5;

export const Rating = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};

export const Score = {
  BAD: 3,
  NORMAL: 5,
  VERY_GOOD: 8,
  AWESOME: 10
};

export const Comment = {
  MIN: 50,
  MAX: 400
};

export const MINUTES_IN_HOUR = 60;
