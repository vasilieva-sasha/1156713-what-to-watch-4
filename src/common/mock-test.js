export const Movie = {
  TITLE: `The Grand Budapest Hotel`,
  GENRE: `Drama`,
  DATE: 2014
};

export const TITLES = [`Fantastic Beasts: The Crimes of Grindelwald`, `Bohemian Rhapsody`, `Macbeth`, `Aviator`,
  `We need to talk about Kevin`, `What We Do in the Shadows`, `Revenant`, `Johnny English`, `Shutter Island`,
  `Pulp Fiction`, `No Country for Old Men`, `Snatch`, `Moonrise Kingdom`, `Seven Years in Tibet`, `Midnight Special`,
  `War of the Worlds`, `Dardjeeling Limited`, `Orlando`, `Mindhunter`, `The Grand Budapest Hotel`];
