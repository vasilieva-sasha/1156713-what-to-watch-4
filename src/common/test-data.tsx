import {Film} from "../types";

export const noop = () => {
  // Mock function for test props
};

export const film: Film = {
  id: 3,
  title: `a`,
  genre: `Drama`,
  releaseDate: 2018,
  runtime: 200,
  poster: `fantastic-beasts-the-crimes-of-grindelwald`,
  posterInfo: `posters/placeimg_218_327_animals`,
  background: `background/placeimg_1000_424_animals`,
  backgroundColor: `rgba`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: {
    score: 8.9,
    count: 240
  },
  text: ``,
  director: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: false
};

export const films: Array<Film> = [
  {
    id: 1,
    title: `a`,
    genre: `Drama`,
    releaseDate: 2018,
    runtime: 200,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`,
    posterInfo: `posters/placeimg_218_327_animals`,
    background: `background/placeimg_1000_424_animals`,
    backgroundColor: `rgba`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.9,
      count: 240
    },
    text: ``,
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    isFavorite: false
  },
  {
    id: 2,
    title: `b`,
    genre: `Drama`,
    releaseDate: 2018,
    runtime: 200,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`,
    posterInfo: `posters/placeimg_218_327_animals`,
    background: `background/placeimg_1000_424_animals`,
    backgroundColor: `rgba`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: {
      score: 8.9,
      count: 240
    },
    text: ``,
    director: `Wes Andreson`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    isFavorite: true
  }
];
