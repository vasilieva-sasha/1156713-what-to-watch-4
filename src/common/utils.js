import {ALL_GENRES, Rating, Score, MINUTES_IN_HOUR} from "./consts";

export const makeLetterUppercase = (string) => {
  return string[0].toUpperCase() + string.slice(1);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getGenres = (films) => {
  return films.reduce((filmGenres, film) => {
    if (!filmGenres.includes(film.genre)) {
      filmGenres.push(film.genre);
    }
    return filmGenres;
  }, [ALL_GENRES]);
};

export const getRatingLevel = (rating) => {
  switch (true) {
    case rating <= Score.BAD:
      return Rating.BAD;
    case rating > Score.BAD && rating <= Score.NORMAL:
      return Rating.NORMAL;
    case rating > Score.NORMAL && rating <= Score.VERY_GOOD:
      return Rating.GOOD;
    case rating < Score.VERY_GOOD && rating < Score.AWESOME:
      return Rating.VERY_GOOD;
    case rating === Score.AWESOME:
      return Rating.AWESOME;
  }
  return null;
};

export const getFilmDuration = (runtime) => {
  const hours = Math.floor(runtime / MINUTES_IN_HOUR);
  const minutes = runtime % MINUTES_IN_HOUR;
  const filmDuration = `${hours}h ${minutes}m`;
  return filmDuration;
};
