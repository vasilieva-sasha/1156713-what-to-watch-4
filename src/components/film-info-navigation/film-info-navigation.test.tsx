import * as React from "react";
import * as renderer from "react-test-renderer";
import FilmInfoNavigation from "./film-info-navigation";
import {noop, film} from "../../common/test-data";


describe(`FilmNavigationComponent`, () => {
  it(`Navigation bar renders correctly`, () => {
    const tree = renderer.create(
        <FilmInfoNavigation film={film}
          detailsScreen={0}
          onLinkClick={noop}/>
    );

    expect(tree).toMatchSnapshot();
  });
});
