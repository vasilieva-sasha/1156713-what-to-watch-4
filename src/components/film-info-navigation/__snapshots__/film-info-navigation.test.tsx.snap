// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmNavigationComponent Navigation bar renders correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.9
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      />
      <span
        className="movie-rating__count"
      >
        240
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Wes Andreson
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Bill Murray
        Edward Norton
        Jude Law
        Willem Dafoe
         and other
      </strong>
    </p>
  </div>
</div>
`;
